merkle/compact/tree.go

```diff
@@ -273,7 +201,12 @@ func (t *Tree) hashes() [][]byte {
 // For a tree whose size is a perfect power of two, only the last entry in
 // nodes is set, and it also matches the tree root hash.
 func (t *Tree) getNodes() [][]byte {
-	n := make([][]byte, len(t.nodes))
-	copy(n, t.nodes)
+	size := t.rng.End()
+	hashes := t.rng.Hashes()
+	n := make([][]byte, bits.Len64(size))
+	for i := len(hashes) - 1; size != 0; i, size = i-1, size&(size-1) {
+		level := bits.TrailingZeros64(size)
+		n[level] = hashes[i]
```

Comment:
Might need a copy here as it looks like it returns data the Range holds. Possibly that should be inside the Range code depending on expected usage.

Reply:
Probably no need to worry about it, as this function is now used only in tests, and I intend to remove it in a follow-up.

Reply:
OK. We should make sure the Range API is clear on this. We don't want unnecessary copying but we also don't want people breaking it.